### PR TITLE
Adding Edge wallet back

### PR DESCRIPTION
**Pull request by chadyo**
I'm working with Edge to get their wallet added back to the bitcoin.org site. They were removed back in September because they didn't respond in time with a request to send an API key to build the site but the issue is now fixed. Please merge these these changes to add edge back to the site. You can tag @paullinator with any questions specific to the fix that was needed to get Edge listed back on bitcoin.org.

### CODE REVIEW

**Comment by crwatkins** [comment]
@chadyo @paullinator Thanks for the submission.  I'm anxious to get Edge re-listed.

However, please note that the source links that you submitted point to an empty organization with no repositories.

**Comment by chadyo** [comment]
Thanks for catching that @crwatkins It was still pointing to Edge's former github url when it was taken off the site. This is now fixed

File: _wallets/edgewallet.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+compat: "mobile android ios"
+level: 3
+platform:
+  - mobile:
+    name: mobile
+    os:
+      - name: android
+        text: "walletedgewallet"
+        link: "https://play.google.com/store/apps/details?id=co.edgesecure.app"
+        source: "https://github.com/EdgeApp"
```
Comment (wbnns) [inline _wallets/edgewallet.md]: This GitHub link needs to point to the repository that contains the source code for the Android app.

File: _wallets/edgewallet.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+          privacy: "checkpassprivacybasic"
+          fees: "checkpassfeecontroldynamic"
+        privacycheck:
+          privacyaddressreuse: "checkpassprivacyaddressrotation"
+          privacydisclosure: "checkfailprivacydisclosureaccount"
+          privacynetwork: "checkfailprivacynetworknosupporttor"
+      - name: ios
+        text: "walletedgewallet"
+        link: "https://itunes.apple.com/us/app/edge-bitcoin-wallet/id1344400091"
+        source: "https://github.com/EdgeApp"
```
Comment (wbnns) [inline _wallets/edgewallet.md]: This GitHub link needs to point to the repository that contains the source code for the iOS app.

**Comment by chadyo** [comment]
Thanks for reviewing @wbnns 

Tagging @paullinator to answer the questions asked above as he would be better to answer them. Once we get the right repo straightened out I can make the requested updates.

**Comment by paullinator** [comment]
@chadyo please change the github links to point to https://github.com/EdgeApp/edge-react-gui

@wbnns the repo above is the full client-side wallet application for both iOS and Android. The API key is necessary to interact with our server side component that only stores client-side encrypted keys and metadata. We have updated the app to have a built-in public API key for personal use and testing. We have also updated the README to correctly specify how to obtain a production API key.

Let us know if you have any other questions.

File: _wallets/edgewallet.md (hunk 1 above)
Comment (chadyo) [reply _wallets/edgewallet.md]: This is now fixed

File: _wallets/edgewallet.md (hunk 2 above)
Comment (chadyo) [reply _wallets/edgewallet.md]: This is now fixed

**Comment by crwatkins** [comment]
@chadyo @paullinator 

This PR specifies (in two places):

`fees: "checkpassfeecontroldynamic"`

Edge allows overriding the fee suggestion, so Edge qualifies for the better description:

`fees: "checkpassfeecontroloverride"`

Other than that, I concur with the scoring in 62f8a9b.

**Comment by chadyo** [comment]
Thanks for the suggestion @crwatkins I've made the change to those two spots

**Comment by crwatkins** [comment]
LGTM fc9a96f

**Comment by wbnns** [comment]
@crwatkins @chadyo Thanks.

@paullinator Thanks for the update. We followed the [updated instructions in the repository](https://github.com/EdgeApp/edge-react-gui#getting-started) to build the iOS app from source, however, upon launching the Simulator we encountered the following error message after the display of the Edge splash screen:

Screen Shot 2019-11-13 at 13 17 03

Please advise.

**Comment by paullinator** [comment]
@wbnns It looks like the react native packager did not launch. You can manually launch it running `yarn start` in a terminal window. Then fully stop and restart the build and simulator.

**Comment by wbnns** [comment]
@paullinator 

Thank you, that resolved the issue.

We took a quick look at the Master Private Key feature in the repository and didn't see the code responsible for generating and storing the keys; it seemed that might be getting done server-side. In other words, it seemed we need to authenticate server-side with an Edge account password before we can retrieve the seed for one or more desired wallets - but that otherwise, that information isn't held locally accessible to the user.

Is that correct?

**Comment by paullinator** [comment]
@wbnns They keys are absolutely being generated and encrypted, and stored client side, with only a backup sent to the server. The repo https://github.com/EdgeApp/edge-core-js implements all of the encryption and backup. It presents an API to edge-react-gui but all of the key generation and encryption is done client side. You can reference that repository to see the code that does the encryption and backup. For each currency there is a client side plugin that does key generation. In the case of Bitcoin it is https://github.com/EdgeApp/edge-currency-bitcoin

File: _translations/en.yml [hunk 3]
```diff
@@ -286,6 +286,7 @@ en:
     walletelectrum: "Electrum's focus is speed and simplicity, with low resource usage. It uses remote servers that handle the most complicated parts of the Bitcoin system, and it allows you to recover your wallet from a secret phrase."
     walletmsigna: "mSIGNA is an advanced yet easy-to-use wallet featuring speed and simplicity, enterprise-level scalability, and strong security. It supports BIP32, multisignature transactions, offline storage, multidevice synchronization, and encrypted electronic and paper backups."
     walletbitcoinwallet: "Bitcoin Wallet is easy to use and reliable, while also being secure and fast. Its vision is de-centralization and zero trust; no central service is needed for Bitcoin-related operations. The app is a good choice for non-technical people."
+    walletedgewallet: "Edge is a Segwit compatible, mobile Bitcoin Wallet making high levels of privacy, security, and decentralization very familiar and usable to the masses. Edge wallets are always automatically encrypted, backed up, and even function when Edge servers go down."
```
Comment (wbnns) [inline _translations/en.yml]: We noted that if the app is unable to reach the auth servers, we are unable to log in and access our wallets. An error message appears:

`Could not reach the auth server: /v2/login`

We recommend that the last portion of the description alluding to wallets functioning even when Edge servers go down be removed, so as to hopefully avoid misunderstandings.

File: _translations/de.yml [hunk 4]
```diff
@@ -182,6 +182,7 @@ de:
...
     walletmsigna: "mSIGNA ist eine fortgeschrittene aber dennoch leicht zu benutzende Wallet, die Geschwindigkeit, Einfachheit, Skalierbarkeit auf Unternehmensebene und ein hohes Maß an Sicherheit bietet. Es wird BIP32, Multisignatur-Transaktionen, Offline-Speicherung, Synchronisierung mit mehreren Geräten sowie verschlüsselte elektronische und physische Backups geboten."
     walletbitcoinwallet: "Bitcoin Wallet ist einfach zu verwenden und zuverlässig, gleichzeitig aber sicher und schnell. Der Fokus ist auf Dezentralisierung und Zero Trust gerichtet: Kein zentraler Dienst wird für Bitcoin-bezogene Aktionen benötigt. Die App ist eine gute Wahl für technisch nicht versierte Personen."
+    walletedgewallet: "Edge ist eine Segwit-kompatible, mobile Bitcoin-Wallet, die Privatsphäre, Sicherheit und Dezentralisierung für jeden zugänglich und nutzbar macht. Edge-Wallets sind immer automatisch verschlüsselt, gesichert und funktionieren auch dann noch, wenn die Edge-Server nicht erreichbar sind."
```
Comment (wbnns) [inline _translations/de.yml]: Please revise or drop change to this file as per comment in: `_translations/en.yml`

File: _translations/hu.yml [hunk 5]
```diff
@@ -176,6 +176,7 @@ hu:
...
     walletmsigna: "Az mSIGNA egy haladó szintű, mégis egyszerűen használható pénztárca. amely gyors, letisztult, cégszintű használatra is alkalmas, valamint nagyon biztonságos. Támogatja a BIP32-t, a csoportos aláírású tranzakciókat, a többkészülékes szinkronizációt, valamint a titkosított elektronikus és papíralapú biztonsági másolatokat."
     walletbitcoinwallet: "A Bitcoin Wallet egyszerűen használható és megbízható, mindemellett pedig biztonságos és gyors. A fejlesztők víziója a decentralizáció és a zéró bizalom; a Bitcoinnal kapcsolatos műveletek elvégzéséhez nem szükséges központi szolgáltatás. Az alkalmazás jó választás fejlesztői háttérrel nem rendelkező felhasználók számára."
+    walletedgewallet: "Az Edge egy Segwit kompatibilis mobile bitcoin tárca ami magas szintű adatvédelmet, biztonságot és decentralizációt garantál, mely nagyon könnyen használható mindenki számára. Az Edge tárca minden esetben automatikusan titkosításra, biztonsági mentésre kerül és még akkor is működik amikor az Edge szerverek nem elérhetőek."
```
Comment (wbnns) [inline _translations/hu.yml]: Please revise or drop change to this file as per comment in: `_translations/en.yml`

File: _translations/id.yml [hunk 6]
```diff
@@ -176,6 +176,7 @@ id:
...
     walletmsigna: "mSIGNA adalah jenis wallet yang canggih namun mudah digunakan, dengan fitur kecepatan dan kesederhanaan, dapat disesuaikan dengan level perusahaan, dan keamanan yang kuat. mSIGNA mendukung BIP32, transaksi dengan tanda tangan berganda, penyimpanan offline, sinkronisasi dengan banyak perangkat, dan pencadangan elektronik maupun kertas yang bersandi."
     walletbitcoinwallet: "Dompet Bitcoin mudah digunakan dan dapat diandalkan, serta aman dan cepat. Visinya adalah desentralisasi dan nol kepercayaan: tidak perlu layanan sentral untuk segala operasi yang berhubungan dengan Bitcoin. Aplikasi ini adalah pilihan yang bagus untuk pengguna awam. "
+    walletedgewallet: "Edge sudah kompatibel untuk segwit, wallet Bitcoin untuk mobile ini mempunyai level privasi tinggi, keamanan dan desentralisasi yang lebih populer untuk digunakan untuk banyak orang. Wallet Edge akan selalu ter-enkripsi, ter-back-up, meskipun server Edge sedang turun. "
```
Comment (wbnns) [inline _translations/id.yml]: Please revise or drop change to this file as per comment in: `_translations/en.yml`

File: _translations/it.yml [hunk 7]
```diff
@@ -180,6 +180,7 @@ it:
...
     walletmsigna: "mSIGNA è un portafoglio avanzato e di facile utilizzo che fornisce velocità e semplicità, scalabilità di livello aziendale e una robusta sicurezza. Supporta BIP32, transazioni multi-firma, archiviazione offline, sincronizzazione su più dispositivi, copie di backup elettroniche cifrate e anche cartacee."
     walletbitcoinwallet: "Bitcoin Wallet è facile da usare e affidabile, pur essendo anche sicuro e veloce. La sua visione è la decentralizzazione e nessun bisogno di fiducia; nessun servizio centrale è necessario per le operazioni riguardanti il Bitcoin. L'app è una buona scelta per gli utenti non tecnici."
+    walletedgewallet: "Edge è un portafoglio mobile di Bitcoin, compatibile con Segwit che rende gli alti livelli di privacy, sicurezza e decentralizzazione molto familiari e utilizzabili da chiunque. I portafogli Edge sono sempre crittografati e sottoposti a backup in modo automatico e funzionano anche quando i server Edge non funzionano."
```
Comment (wbnns) [inline _translations/it.yml]: Please revise or drop change to this file as per comment in: `_translations/en.yml`

File: _translations/ja.yml [hunk 8]
```diff
@@ -244,6 +244,7 @@ ja:
     walletelectrum: "Electrumの特徴は、低リソース、高速、シンプルな点です。ビットコインシステムの最も複雑な部分を扱うリモート・サーバーを使用しており、秘密の復元フレーズを使用してビットコイン・ウォレットを復元する事ができます。"
     walletmsigna: "mSIGNAは高性能かつ簡単に使用できるウォレットで、高速性、シンプルさ、企業レベルの拡張性、セキュリティーの強さが特徴です。これは、BIP32、マルチシグトランザクション、オフラインストレージ、マルチデバイス間の同期、暗号化された電子的および紙面によるバックアップをサポートしています。"
     walletbitcoinwallet: "ビットコインウォレットは、使用が簡単で信頼性があり、安全かつ高速な取引ができます。そのビジョンは、非中央集権化とゼロトラストにあります。つまり、ビットコインの運用に、中央集権的なサービスは必要ないと言う事です。このアプリは、技術的な知識を持たない方にも適しています。"
+    walletedgewallet: "Edgeは高レベルのプライバシー、セキュリティ、非中央集権性を備える、とても使いやすく、Segwitと互換性のあるビットコインモバイルウォレットです。Edgeウォレットは自動的に暗号化やバックアップがなされ、Edgeサーバーが停止しても機能します。"
```
Comment (wbnns) [inline _translations/ja.yml]: Please revise or drop change to this file as per comment in: `_translations/en.yml`

File: _translations/nl.yml [hunk 9]
```diff
@@ -182,6 +182,7 @@ nl:
...
     walletmsigna: "mSIGNA is een geavanceerde maar makkelijk te gebruiken wallet die gekenmerkt wordt door snelheid en eenvoud, schaalbaarheid op ondernemingsniveau, en een sterke beveiliging. Het ondersteunt BIP32, transacties met meerdere handtekeningen, offline opslag, synchronisatie op meerdere apparaten, en versleutelde elektronische en papieren back-ups. "
     walletbitcoinwallet: "Bitcoin Wallet is eenvoudig te gebruiken en betrouwbaar, terwijl het ook veilig en snel is. De visie is decentralisatie en \"zero trust\"; er is geen centrale dienst nodig voor de Bitcoin-gerelateerde handelingen. De applicatie is een goede keuze voor niet-technische personen."
+    walletedgewallet: "Edge is een mobiele Bitcoin-portemonnee met Segwit-ondersteuning die hoge niveaus van privacy, beveiliging en decentralisatie zeer vertrouwd en bruikbaar maakt voor de massa. Edge-portemonnees worden altijd automatisch versleuteld, gebackupt, en functioneren zelfs wanneer de Edge-servers uitvallen."
```
Comment (wbnns) [inline _translations/nl.yml]: Please revise or drop change to this file as per comment in: `_translations/en.yml`

File: _translations/pt_BR.yml [hunk 10]
```diff
@@ -181,6 +181,7 @@ pt_BR:
...
     walletmsigna: "mSIGNA é uma carteira avançada mas fácil de usar, com rapidez e simplicidade, escalabilidade de nível empresarial e uma forte segurança. Ela suporta BIP32, transações multi-assinatura, armazenamento off-line, sincronização multidispositivo e backups criptografados eletrônicos e em papel."
     walletbitcoinwallet: "Bitcoin Wallet é fácil de usar e confiável, sendo também seguro e rápido. Sua visão é descentralização e confiança zero; nenhum serviço central é necessário para operações relacionadas com Bitcoin. O aplicativo é uma boa escolha para pessoas não técnicas."
+    walletedgewallet: "Edge é uma carteira móvel de Bitcoin compatível com Segwit, que aumenta o nível de privacidade, segurança e descentralização muito familiares e utilizado pelas pessoas. As carteiras Edge são criptografadas sempre  automaticamente, armazenadas em backup e funcionam até mesmo quando os servidores Edge ficam inativos."
```
Comment (wbnns) [inline _translations/pt_BR.yml]: Please revise or drop change to this file as per comment in: `_translations/en.yml`

**Comment by paullinator** [comment]
@wbnns  As long as you have logged in at least one time on your device, you will still be able to login even if the auth servers are unavailable. PIN login will not work, but biometric and full password logins do. After login, users can send and receive funds as normal as the application talks directly to the open blockchain network

**Comment by wbnns** [comment]
@paullinator 

Understood, however, there are no tool tips or other UI elements that provide the guidance you've given us, to the user. By default, the app prompts for a PIN if fingerprint login isn't available, which will generate an error message upon being entered if the platform is unavailable. Additionally, if a user has valid credentials (username and password), but hasn't logged in on a device before, they will encounter an error message if the platform is unavailable and won't be able to access their wallets using the app.

For the latter scenario, when creating an account, the user is shown their username/password/PIN and encouraged to safely store them so that their account can be accessed, however, there is no onboarding process for looking up and storing the wallet seed. This might give a false sense of security if the platform became unavailable, since the username/password/PIN wouldn't be able to access and decrypt the wallets server-side.

Since it seems likely that users could experience non-trivial issues with the app if your servers were to go down, we still recommend that the claim stating otherwise be removed.

**Comment by paullinator** [comment]
@wbnns Thanks for the feedback and your comments are valid. We will remove the referenced statement and update the PR. Thanks.

**Comment by chadyo** [comment]
Checked in with the revised wallet description.

**Pull request merged**